src/main/java/programminglife/ProgrammingLife.java

```diff
@@ -69,23 +74,41 @@ public void start(Stage stage) throws Exception {
 
         vbox = FXMLLoader.load(getClass().getResource("/Basic_Gui.fxml"));
         primaryStage.setScene(new Scene(vbox, 1000, 900));
-        primaryStage.setOnCloseRequest(e -> close());
+        primaryStage.setOnCloseRequest(confirmCloseEventHandler);
+        Button close = new Button("Close Application");
+        close.setOnAction(event -> primaryStage.fireEvent(
+                new WindowEvent(primaryStage, WindowEvent.WINDOW_CLOSE_REQUEST))
+        );
         primaryStage.sizeToScene();
         primaryStage.show();
     }
 
+    private EventHandler<WindowEvent> confirmCloseEventHandler = event -> {
```

Comment:
This event is from the cross (on windows) / red dot (on mac) to stop the application. Therefore I think it should be handled in the main. The quit menuItem created by the GUI however is handled in the controller